conans/model/info.py

```diff
@@ -557,6 +557,8 @@ def package_id(self):
         result.append(requires_sha)
         if self.python_requires:
             result.append(self.python_requires.sha)
+        if hasattr(self, "conf"):
```

Comment:
This is the only change regarding package_id so far.
Recipe will do (in test):
```python
def package_id(self):
     self.info.conf = self.conf
```
It is still missing all the methods for limiting which vars are used, which modules, do a copy, etc
Also we need to talk about the ``conaninfo.txt`` for ``conf``, but maybe not necessary at this first iteration (we could say and document that conf still not affect package_id)